task/bq2bq/executor/bumblebee/transformation.py

```diff
@@ -507,20 +512,17 @@ def execute(self, tasks: List[T]):
         if tasks is not None and len(tasks) > 0:
             self._concurrent_execute_task(tasks, self.concurrency)
 
+    def execute_task(self, task):
+        return task.execute()
+
     # TODO: future should check for task exception
     def _concurrent_execute_task(self, tasks, concurrency: int):
-        async def execute_all():
-            batches = split_list(tasks, concurrency)
-            for batch in batches:
-                futures = [task.async_execute() for task in batch]
-                await asyncio.wait(futures)
-
-        self.async_exec(execute_all)
-
-    def async_exec(self, func):
-        ioloop = asyncio.get_event_loop_policy().new_event_loop()
-        ioloop.run_until_complete(func())
-        ioloop.close()
+
+        with ThreadPoolExecutor(concurrency) as executor:
+            threads = {executor.submit(self.execute_task, task): task for task in tasks}
```

Comment:
minor comment i believe we are the threads variable is a map of future & task, variable name can be future_map or just futures